fourcc.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import sys
+
+cc = os.getenv("CC", "cc")
+pkg_config = os.getenv("PKG_CONFIG", "pkg-config")
+
+cflags = subprocess.check_output([pkg_config, "libdrm", "--cflags-only-I"]).decode().strip().split()
+if not cflags[0].startswith("-I"):
+	raise Exception("Expected one include dir for libdrm")
+libdrm_include = cflags[0][2:]
+fourcc_include = libdrm_include + "/drm_fourcc.h"
+
+ident_list = []
+with open(fourcc_include) as input_file:
+	for l in input_file.readlines():
+		parts = l.split()
+		if len(parts) < 3 or parts[0] != "#define":
+			continue
+		ident = parts[1]
+		if not ident.startswith("DRM_FORMAT_") and not ident.startswith("I915_FORMAT_"):
+			continue
+		if "(" in ident:
+			# Some complicated formats use a macro with parameters
+			continue
+		ident_list.append(ident)
+
+c_file_name = "tables.h"
```

Comment:
This should probably be read from `argv[1]`

Reply:
I was thinking the same but not sure if it was worth the effort.